File: C++/题目1419：文献排序.cpp

#include <stdio.h>
#include <string.h>
#include <string>
#include<algorithm>
using namespace std;
const int maxn = 102;
const int maxm = 202;
int n,i;
struct Node{
	char allStr[maxm];	
}nodes[maxn];

void toLowCase(char s[]){
	int len = strlen(s);
	for(i = 0 ; i < len ; i++){
		if(s[i] >= 'A' && s[i] <= 'Z'){
			s[i] = s[i] - 'A' + 'a';
		}
	}
}
  
bool cmp(Node node1, Node node2){
	toLowCase(node1.allStr);
	toLowCase(node2.allStr);
	return strcmp(node1.allStr,node2.allStr) < 0;
}

int main(){

	while(scanf("%d",&n)!=EOF)
    {
        gets(nodes[0].allStr);
        for(i = 0;i < n; i++){
			gets(nodes[i].allStr);
		}
        sort(nodes,nodes+n,cmp);
        for(i = 0;i < n; i++){
			puts(nodes[i].allStr);
		}
    }
	return 0;
}
/**************************************************************
	Problem: 1419
	User: wangzhenqing
	Language: C++
	Result: Accepted
	Time:90 ms
	Memory:1044 kb
****************************************************************/


                        

File: Java/题目1419：文献排序.java

// 题目1419：文献排序

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:55
 * @url:http://ac.jobdu.com/problem.php?pid=1419
 * 解题思路参考csdn:http://blog.csdn.net/u013027996/article/details/17231173
 */

import java.util.Arrays;
import java.util.Scanner;

public class Main {
    /*
     * 1419
     */
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        while (scanner.hasNextLine()) {
            int N = Integer.parseInt(scanner.nextLine());
            SortString array[] = new SortString[N];
            for (int i = 0; i < N; i++) {
                String ab = scanner.nextLine();
                SortString sortStr = new SortString(ab.toLowerCase(), ab);
                array[i] = sortStr;
            }
            Arrays.sort(array);
            for (int i = 0; i < N; i++) {
                System.out.println(array[i].getAllMsg());
            }
        }
    }

    static class SortString implements Comparable<SortString> {

        private String firStr;
        private String allMsg;

        public String getFirStr() {
            return firStr;
        }

        public void setFirStr(String firStr) {
            this.firStr = firStr;
        }

        public String getAllMsg() {
            return allMsg;
        }

        public void setAllMsg(String allMsg) {
            this.allMsg = allMsg;
        }

        public SortString(String firStr, String allMsg) {
            super();
            this.firStr = firStr;
            this.allMsg = allMsg;
        }

        public int compareTo(SortString o) {
            return this.getFirStr().compareTo(o.getFirStr());
        }

    }
}
/**************************************************************
 * Problem: 1419
 * User: wzqwsrf
 * Language: Java
 * Result: Accepted
 * Time:1450 ms
 * Memory:30228 kb
 ****************************************************************/


                        


